routes/dao_coin_exchange_with_fees.go

```diff
@@ -74,23 +72,47 @@ func ValidateTradingFeeMap(feeMap map[string]uint64) error {
 	return nil
 }
 
+func IsExtraDataValueTrueString(extraData map[string]string, key string) bool {
+	newMap := make(map[string][]byte)
+	for k, v := range extraData {
+		newMap[k] = []byte(v)
+	}
```

Comment:
It seems a little unnecessary to make a copy of the entire map. If there are a lot of KVs in the extra data, this may be slow. Can we just make a copy of the value we need?